src/tritonserver.cc

```diff
@@ -2725,4 +2725,132 @@ TRITONSERVER_ServerInferAsync(
   return nullptr;  // Success
 }
 
+//
+// TRITONSERVER_MetricFamily
+//
+TRITONSERVER_Error*
+TRITONSERVER_MetricFamilyNew(
+    TRITONSERVER_MetricFamily** family, TRITONSERVER_MetricKind kind,
+    const char* name, const char* description, void* registry_ptr)
+{
+#ifdef TRITON_ENABLE_METRICS
+  const auto& registry =
+      reinterpret_cast<std::shared_ptr<prometheus::Registry>*>(registry_ptr);
+  // TODO: Remove this after review
+  // TODO: Verify passing registry_ptr from API caller is OK.
+  //       Calling GetRegistry() here did not add the new metrics to registry.
+  // const auto& registry = tc::Metrics::GetRegistry();
+  *family = reinterpret_cast<TRITONSERVER_MetricFamily*>(
+      new tc::MetricFamily(kind, name, description, *registry));
+  return nullptr;  // Success
+#else
+  return TRITONSERVER_ErrorNew(
+      TRITONSERVER_ERROR_UNSUPPORTED, "metrics not supported");
+#endif  // TRITON_ENABLE_METRICS
+}
+
+TRITONSERVER_Error*
+TRITONSERVER_MetricFamilyDelete(TRITONSERVER_MetricFamily* family)
+{
+#ifdef TRITON_ENABLE_METRICS
+  delete reinterpret_cast<tc::MetricFamily*>(family);
+  return nullptr;  // Success
+#else
+  return TRITONSERVER_ErrorNew(
+      TRITONSERVER_ERROR_UNSUPPORTED, "metrics not supported");
+#endif  // TRITON_ENABLE_METRICS
+}
+
+//
+// TRITONSERVER_Metric
+//
+TRITONSERVER_Error*
+TRITONSERVER_MetricNew(
+    TRITONSERVER_Metric** metric, TRITONSERVER_MetricFamily* family,
+    const TRITONSERVER_Parameter** labels, const uint64_t label_count)
+{
+#ifdef TRITON_ENABLE_METRICS
+  std::vector<const tc::InferenceParameter*> labels_vec;
+  for (size_t i = 0; i < label_count; i++) {
+    labels_vec.emplace_back(
+        reinterpret_cast<const tc::InferenceParameter*>(labels[i]));
+  }
+  *metric = reinterpret_cast<TRITONSERVER_Metric*>(
```

Comment:
try-catch here as well? Or may be not as it won't fail for a valid family with a kind.